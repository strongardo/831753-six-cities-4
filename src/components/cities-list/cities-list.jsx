import React from "react";
import {connect} from "react-redux";
import {bindActionCreators} from "redux";
import {createCityAction, createOffersAction} from "../../reducer.js";
import PropTypes from "prop-types";

const CitiesList = (props) => {
  const getCityClassName = (city) => {
    return props.city === city
      ? `locations__item-link tabs__item tabs__item--active`
      : `locations__item-link tabs__item`;
  };

  const handleCityClick = (city) => {
    props.createCityAction(city.name);
    props.createOffersAction(city.name);
  };

  const getCitiesMarkup = () => {
    return props.cities.map((city) => {
      return (
        <li key={city.id} className="locations__item">
          <a className={getCityClassName(city.name)} onClick={() => handleCityClick(city)}>
            <span>{city.name}</span>
          </a>
        </li>
      );
    });
  };

  return (
    <section className="locations container">
      <ul className="locations__list tabs__list">
        {getCitiesMarkup()}
      </ul>
    </section>
  );
};

CitiesList.propTypes = {
  city: PropTypes.string.isRequired,
  cities: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string,
    id: PropTypes.number,
  })).isRequired,
  createCityAction: PropTypes.func.isRequired,
  createOffersAction: PropTypes.func.isRequired,
};

const matchDispatchToProps = (dispatch) => {
  return bindActionCreators({createCityAction, createOffersAction}, dispatch);
};

export {CitiesList};
export default connect(null, matchDispatchToProps)(CitiesList);
