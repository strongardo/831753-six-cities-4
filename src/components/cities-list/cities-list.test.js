import React from "react";
import renderer from "react-test-renderer";
import {CitiesList} from "./cities-list.jsx";

const cities = [
  {
    name: `Paris`,
    id: 1,
  },
];

it(`Should CitiesList render correctly`, () => {
  const tree = renderer
    .create(<CitiesList
      city={`Paris`}
      cities={cities}
      createCityAction={() => {}}
      createOffersAction={() => {}}
    />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
