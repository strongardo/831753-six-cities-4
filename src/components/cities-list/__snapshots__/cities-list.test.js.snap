// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should CitiesList render correctly 1`] = `
<section
  className="locations container"
>
  <ul
    className="locations__list tabs__list"
  >
    <li
      className="locations__item"
    >
      <a
        className="locations__item-link tabs__item tabs__item--active"
        onClick={[Function]}
      >
        <span>
          Paris
        </span>
      </a>
    </li>
  </ul>
</section>
`;
